File: src/main/java/cc/sukazyo/cono/morny/util/FileUtils.java
package cc.sukazyo.cono.morny.util;

import javax.annotation.Nonnull;
import java.io.FileInputStream;
import java.io.IOException;
import java.math.BigInteger;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class FileUtils {
	
	@Nonnull
	public static String getMD5Three (@Nonnull String path) throws IOException, NoSuchAlgorithmException {
		final BigInteger bi;
		final byte[] buffer = new byte[8192];
		int len;
		final MessageDigest md = MessageDigest.getInstance("MD5");
		final FileInputStream fis = new FileInputStream(path);
		while ((len = fis.read(buffer)) != -1) {
			md.update(buffer, 0, len);
		}
		fis.close();
		final byte[] b = md.digest();
		bi = new BigInteger(1, b);
		return bi.toString(16);
	}
	
}

File: morny-core/src/main/scala/cc/sukazyo/cono/morny/util/FileUtils.scala
package cc.sukazyo.cono.morny.util

import java.io.{FileInputStream, IOException}
import java.security.{MessageDigest, NoSuchAlgorithmException}
import scala.util.Using

/**
  * @todo docs
  * @todo some tests?      
  */
object FileUtils {
	
	@throws[IOException|NoSuchAlgorithmException]
	def getMD5Three (path: String): String = {
		val buffer = Array.ofDim[Byte](8192)
		var len = 0
		val algo = MessageDigest.getInstance("MD5")
		Using (FileInputStream(path)) { stream =>
			len = stream.read(buffer)
			while (len != -1)
				algo.update(buffer, 0, len)
				len = stream.read(buffer)
		}
		import cc.sukazyo.cono.morny.system.utils.ConvertByteHex.toHex
		algo.digest toHex
	}
	
}
